documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,1243 @@
+the client can connect again by calling `connect`.
+
+In the v2 SDK, an MQTT5 client can stop a session by calling the
+[stop](https://awslabs.github.io/aws-crt-python/api/mqtt5.html#awscrt.mqtt5.Client.stop) method.
+You can provide an optional [DisconnectPacket](https://awslabs.github.io/aws-crt-python/api/mqtt5.html#awscrt.mqtt5.DisconnectPacket)
+parameter. A closed client can be started again by calling
+[start](https://awslabs.github.io/aws-crt-python/api/mqtt5.html#awscrt.mqtt5.Client.start).
+
+
+#### Example of disconnecting a client in the v1 SDK
+
+```python
+client.disconnect();
+
+```
+
+#### Example of disconnecting a client in the v2 SDK
+
+```python
+mqtt5_client.stop(
+        disconnect_packet=mqtt5.DisconnectPacket(
+                reason_code=mqtt5.*DisconnectReasonCode**.**NORMAL_DISCONNECTION**,
+                session_expiry_interval_sec=3600*))
+
+```
+
+### Reconnects
+
+The v1 SDK attempts to reconnect automatically using a [Progressive Reconnect Back Off](https://github.com/aws/aws-iot-device-sdk-python/blob/master/README.rst#progressive-reconnect-back-off)
+until connection succeeds or `client.disconnect()` is called.
+
+The v2 SDK attempts to reconnect automatically until the connection succeeds or `client.stop()` is called.
+The reconnection parameters, such as min/max delays and
+[jitter modes](https://awslabs.github.io/aws-crt-python/api/mqtt5.html#awscrt.mqtt5.ExponentialBackoffJitterMode),
+are configurable through [awsiot.mqtt5\_client\_builder](https://aws.github.io/aws-iot-device-sdk-python-v2/awsiot/mqtt5_client_builder.html#module-awsiot.mqtt5_client_builder).
+
+
+#### Example of tweaking reconnection settings in v1
+
+```python
+baseReconnectQuietTimeSecond = 1 # Initial backoff time
+maxReconnectQuiteTimeSecond = 23 # maximum backoff time
+stableConnectionTimeSecond = 20  # the time the connection is considered stable
+client``.``configureAutoReconnectBackoffTime``(
+        ``baseReconnectQuietTimeSecond``,
+        ``maxReconnectQuiteTimeSecond``,
+        ``stableConnectionTimeSecond``)
+
+```
+
+#### Example of tweaking reconnection settings in the v2 SDK
+
+```python
+mqtt5_client = mqtt5_client_builder.mtls_from_path(
+        endpoint="<prefix>-ats.iot.<region>.amazonaws.com",
+        cert_filepath="<certificate file path>",
+        pri_key_filepath="<private key file path>",
+ ...
+        max_reconnect_delay_ms=200,
+        min_reconnect_delay_ms=200,
+        min_connected_time_to_reset_reconnect_delay_ms=200,
+        retry_jitter_mode=mqtt5.ExponentialBackoffJitterMode.FULL)
+
+mqtt5_client.start()
+
+```
+
+### Offline Operations Queue
+
+In the v1 SDK, if you're having too many in-flight QoS 1 messages, you can encounter the `too many publishes in Progress` error
+on publishing messages. This is caused by the so-called in-flight publish limit. By default, ther v1 SDK supports
+a maximum of 20 in-flight operations.
```

Comment:
Is it true for python sdk?

Reply:
we also have it for python, it is hidden in the source code though